internal/gcs/guestconnection.go

```diff
@@ -123,18 +122,24 @@ func (gc *GuestConnection) connect(ctx context.Context, isColdStart bool, initGu
 		MaximumVersion: protocolVersion,
 	}
 	var resp negotiateProtocolResponse
-	resp.Capabilities.GuestDefinedCapabilities = &gc.caps
 	err = gc.brdg.RPC(ctx, rpcNegotiateProtocol, &req, &resp, true)
 	if err != nil {
 		return err
 	}
 	if resp.Version != protocolVersion {
 		return fmt.Errorf("unexpected version %d returned", resp.Version)
 	}
+
 	gc.os = strings.ToLower(resp.Capabilities.RuntimeOsType)
 	if gc.os == "" {
 		gc.os = "windows"
 	}
+
+	gc.caps, err = unmarshalGuestCapabilities(gc.os, resp.Capabilities.GuestDefinedCapabilities)
```

Comment:
Can we implement the `MarshalJSON`/`UnmarshalJSON` methods on the `gcsCapabilities` type so that we don't have to explicitly unmarshal it here?

Reply:
Discussed together offline. We decided to not implement this right now since GuestDefinedCapabilities is an interface and other options seemed unnecessary.